Automattic/jetpack
# Pages: Add quick actions to change the homepage and the posts page

Part of https://github.com/Automattic/wp-calypso/issues/95620

## Proposed changes:

In order to remove the Pages screen from Calypso, we need to port some missing features to WP Admin first.

One of these features is the ability to change the homepage and the posts page directly from the pages list:

<img width="400" alt="Screenshot 2024-12-19 at 11 01 11" src="https://github.com/user-attachments/assets/381ec510-7509-4d68-af30-c56128d9f5ea" />



This PR implements it in WP Admin as quick actions displayed in the pages list:

<img width="566" alt="Screenshot 2024-12-19 at 13 47 52" src="https://github.com/user-attachments/assets/12d38143-491f-4381-9f6b-456c60a4a871" />


### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

- Apply these changes to your WP.com site
- Go to `/wp-admin/options-reading.php` and note the current settings for the homepage/posts page.
- Go to `/wp-admin/edit.php?post_type=page`
- Make sure the quick actions are available (✅) or not (❌) according to the following conditions. Basically, if you don't see them in Calypso, you shouldn't get them in WP Admin; if you don't see them in Calypso, you shouldn't get them in WP Admin.

Scenario | Set as homepage | Set as posts page | Unset as posts page 
--- | --- | --- | ---
Homepage: Latest posts | ❌ | ❌ | ❌
Homepage: Static page – None selected<br>Posts page: None selected | ❌ | ❌ | ❌
Homepage: Static page – None selected<br>Posts page: Something selected | ❌ | ❌ | ❌
Homepage: Static page – Something selected<br>Posts page: None selected  | ✅ (\*)<br><sub><sup>(\*) Except on the current homepage</sub></sup> | ✅ | ❌
Homepage: Static page – Something selected<br>Posts page: Something selected  | ✅ (\*)<br><sub><sup>(\*) Except on the current homepage</sub></sup> | ✅ (\*)<br><sub><sup>(\*) Except on the current post page</sub></sup> | ✅ (\*)<br><sub><sup>(\*) Only on the current post page</sub></sup>

- Make sure you can use the quick actions to change the homepage/posts page settings.
